datamonster_api/tests/lib/regression.py

```diff
@@ -0,0 +1,160 @@
+import datetime
+import numpy
+import pandas
+import pytest
+
+from datamonster_api import DataMonster, Aggregation, DataMonsterError
+from regression_keys import DM_API_KEY_ID, DM_API_SECRET
+
+
+dm = DataMonster(DM_API_KEY_ID, DM_API_SECRET, server="http://staging.adaptivemgmt.com")
+
+
+def assert_data_frame(df, length):
+    assert all(
+        df.columns
+        == [u"dimensions", u"end_date", u"start_date", u"value", u"time_span"]
+    )
+    assert all(
+        df.dtypes.values
+        == [
+            numpy.dtype("O"),
+            numpy.dtype("<M8[ns]"),
+            numpy.dtype("<M8[ns]"),
+            numpy.dtype("float64"),
+            numpy.dtype("<m8[ns]"),
+        ]
+    )
+    assert len(df) == length
+
+
+def assert_frame_equal(df1, df2):
+    from pandas.util.testing import assert_frame_equal
+
+    assert_frame_equal(
+        df1.reset_index(drop=True).sort_index(axis=1),
+        df2.reset_index(drop=True).sort_index(axis=1),
+    )
+
+
+def test_company():
+    company = dm.get_company_by_id(335)
+    assert company.pk == 335
+    assert company.name == "AMAZON"
+    assert company.ticker == "AMZN"
+    assert company.section_type == "Company"
+    amzn_data_sources = set(company.datasources)
+    amzn_data_source = dm.get_datasource_by_id("cd924848-5c49-4622-95a7-ee6d2cfe24b7")
+    assert amzn_data_source.name in {i.name for i in amzn_data_sources}
+    assert len(amzn_data_sources) == 241
+
+    company = dm.get_company_by_id(1257)
+    assert company.name == "MASTERCARD SECTOR INSIGHTS"
+    assert company.section_type == "Macro"
+
+
+def test_aggregation():
+    company = dm.get_company_by_id(335)
+    agg = Aggregation(period="week", company=company)
+    assert agg.period == "week"
+    assert agg.company == company
+
+
+def test_data_source():
+    """ Test EarnestQuery Panel Sales
+    """
+    company = dm.get_company_by_ticker("W")
+    assert company.name == "WAYFAIR"
+    assert company.ticker == "W"
+    assert company.pk == 128
+
+    ds = next(
+        dm.get_datasources(query="1010data Blended Credit & Debit Sales Index YoY")
+    )
+    assert ds.name == "1010data Blended Credit & Debit Sales Index YoY"
+    assert ds._id == "3de84b2e-604f-4ea7-901f-61601eef8e0e"
+    assert ds.category == "Blended Payment Data"
+    assert len(list(ds.companies)) == 190
+
+    df = ds.get_data(company, end_date="2019-09-09")
```

Comment:
Earnest is a bad choice for regression because they change historical data on future loads. I know 1010 data doesn't do this, so they would be a better data source for this.

Reply:
good to know, updated